torch/csrc/cudnn/Conv.cpp

```diff
@@ -142,17 +179,30 @@ size_t getMaxWorkspaceSize(cudnnHandle_t handle, const Convolution& conv, algo_t
     return max_ws_size;
 }
 
+template<typename perf_t>
+perf_t getBestAlgorithm(perf_t *perfResults, bool deterministic, int n_algo) {
+  if (deterministic) {
+    for (int i = 0; i < n_algo; i++) {
+      if (perfResults[i].status == CUDNN_STATUS_SUCCESS && perfResults[i].determinism == CUDNN_DETERMINISTIC) {
```

Comment:
the algorithms are returned in sorted way by cudnn i.e. first is the fastest etc.
if there was error in benchmarking, we fallback to default algo which are also deterministic and could be a better choice to make